script.js

```diff
@@ -0,0 +1,449 @@
+                    obj.ingredients.forEach((ingredient) => {
+                        const listItem = document.createElement('li');
+                        listItem.innerText = ingredient.amount > 0 
+                            ? `${ingredient.amount || ""} ${ingredient.unit || ""} of ${ingredient.name}`
+                            : `${ingredient.name}: to taste`;
+                        ingredientsList.appendChild(listItem);
+                    });
+                    recipeCard.appendChild(ingredientsList);
+
+                    const recipeDescription = document.createElement('p');
+                    recipeDescription.innerText = `Description: ${obj.description}`;
+                    recipeDescription.classList.add('description'); 
+                    recipeCard.appendChild(recipeDescription);
+
+                    const collapseButton = document.createElement('button');
+                    collapseButton.innerText = 'Collapse';
+                    collapseButton.classList.add('collapse-button'); 
+                    recipeCard.appendChild(collapseButton);
+
+                    // Make a card collapse
+                    collapseButton.addEventListener("click", (e) => {
+                        e.stopPropagation(); // Prevent triggering the card click event
+                        ingredientsList.remove();
+                        recipeDescription.remove();
+                        collapseButton.remove();
+                    });
+        }
+            })
+
+    
+    })
+}
+
+//rendering all recipes
+renderRecipeCard(recipesContainer, allRecipes);
+
+//find recipes by number of ingredients
+const getRecipeByIngredientNbr = (num)=>{
+    const foundRecipes = [];
+    allRecipes.forEach((obj)=>{
+        if(obj.ingredients.length===num) {
+            foundRecipes.push(obj);
+        }
+    })
+    return foundRecipes;
+}
+
+
+ingredientSearchBtn.addEventListener("click", (e)=> {
+    const num = Math.sqrt((parseInt(ingredientNumber.value) ** 2)); // because I'm lazy I am getting the square root of the square of the input, to make sure it's not a negative value
+    result = getRecipeByIngredientNbr(num);
+    renderRecipeCard(recipesSearchResults, result);
+
+})
+
+
+
+// finding recipes by keywords
+const getRecipeByKeyword = (array, str) =>{
+    const foundRecipes = [];
+    //If I understand regex correctly, this is supposed to catch anything that is not a space or a letter
+    const regex = /[^a-zA-Z\s]/g; 
+    const cleanStr = str.replace(regex, "").toLowerCase();
+    //handling empty inputs
+    if (!cleanStr.trim()) { 
+        alert("Please enter a valid keyword.");
+        return;  
+    };
+    array.forEach((obj)=>{
+        let found = false;
+        for (const key in obj) {
+            const value = obj[key];
+            // check for each value if it's a string 
+            if (typeof value === "string" && value.includes(cleanStr)){
```

Comment:
I would only check the type of the value if we don't trust how obj map was created. In this case, it was created by us, no?
If so, then the first check can be omitted